spacy_llm/tests/tasks/test_entity_linker.py

```diff
@@ -0,0 +1,753 @@
+    """Tests highlighting of entities in text."""
+    nlp = spacy.blank("en")
+    text = "Alice goes to Boston to see the Boston Celtics game."
+    doc = nlp.make_doc(text)
+    doc.ents = [
+        Span(doc=doc, start=3, end=4, label="LOC"),
+        Span(doc=doc, start=7, end=9, label="ORG"),
+    ]
+
+    assert (
+        EntityLinkerTask.highlight_ents_in_text(doc)
+        == "Alice goes to *Boston* to see the *Boston Celtics* game."
+    )
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+@pytest.mark.parametrize("loader", ("yaml", "serialized_no_nlp"))
+def test_entity_linker_predict_alternative_kb_inits(loader, request, tmp_path):
+    """Use OpenAI to get EL results with different KB initialization methods."""
+    cfg = request.getfixturevalue("fewshot_cfg_string")
+    if loader == "serialized_no_nlp":
+        overrides = {
+            "paths.el_kb": str(tmp_path / "entity_linker" / "kb"),
+            "paths.el_desc": str(tmp_path / "desc.csv"),
+        }
+        build_el_pipeline(nlp_path=tmp_path, desc_path=tmp_path / "desc.csv")
+    else:
+        overrides = {
+            "paths.el_kb": str(
+                Path(__file__).resolve().parent / "misc" / "el_kb_data.yml"
+            ),
+        }
+    orig_config = Config().from_str(cfg, overrides=overrides)
+    if loader == "yaml":
+        orig_config["initialize"]["components"]["llm"]["candidate_selector"][
+            "kb_loader"
+        ]["@llm_misc"] = "spacy.KBFileLoader.v1"
+        orig_config["initialize"]["components"]["llm"]["candidate_selector"][
+            "kb_loader"
+        ].pop("nlp_path")
+        orig_config["initialize"]["components"]["llm"]["candidate_selector"][
+            "kb_loader"
+        ].pop("desc_path")
+
+    nlp = spacy.util.load_model_from_config(orig_config, auto_fill=True)
+    nlp.initialize(lambda: [])
+
+    text = "Alice goes to Boston to see the Boston Celtics game."
+    doc = nlp.make_doc(text)
+    doc.ents = [
+        Span(doc=doc, start=3, end=4, label="LOC"),  # Q100
+        Span(doc=doc, start=7, end=9, label="ORG"),  # Q131371
+    ]
+    doc = nlp(doc)
+    assert len(doc.ents) == 2
+    assert doc.ents[0].kb_id_ == "Q100"
+    assert doc.ents[1].kb_id_ == "Q131371"
+
+
+@pytest.mark.external
+@pytest.mark.skipif(has_openai_key is False, reason="OpenAI API key not available")
+def test_init_with_code():
+    """Tests EntityLinker task example with only code, no config."""
+    nlp = spacy.blank("en")
+    llm_ner = nlp.add_pipe("llm_ner")
+    for label in ("PERSON", "ORGANISATION", "LOCATION", "SPORTS TEAM"):
+        llm_ner.add_label(label)
+    nlp.initialize()
+
+    nlp.add_pipe("llm_entitylinker")._task.set_candidate_selector(
+        KBCandidateSelector(
+            kb_loader=KBFileLoader(
+                path=Path(__file__).parent / "misc" / "el_kb_data.yml"
+            ),
+            top_n=5,
+        ),
+        nlp.vocab,
+    )
```

Comment:
The fact that this test fails when you call `nlp.initialize()` _after_ adding the `llm_entitylinker` pipe is a clear red flag that shouldn't be ignored. The general approach in spaCy is that you first create/define all your components, then call `nlp.initialize()` on the entire pipeline. We need to make sure that this new component adheres to that contract.

This will need more work in the `entity_linker` task. Basically we should only error when the EL initializes itself and can not find a candidate selector either in the `initialize` arguments or because it was set previously.

Reply:
Updated in https://github.com/explosion/spacy-llm/pull/241/commits/7770a13cc8a6ed54f104e8b8b2334c9a561f797c.